### DISCUSSION TITLE
How to overwrite spacing of ScalarImage ?

### DISCUSSION THREAD

**Discussion by Optimox** [TorchIO-project/torchio#828] [Q&A] [answered]
Hi,

I have several 3D stacks and annotations that can be either in dicom or tiff format.

The spacing are not necessarily correct inside the dicom or tiff files so when I try to access data with augmentation from the dataloader I end up with this error:

```
RuntimeError: More than one value for "spacing" found in subject images:
{'image': (0.001002435076546, 0.001, 0.0010000000474974491),
 'mask': (1.0000000150184933, 1.0000000150184933, 1.0)}
 ```

But I don't want to Resample my data as I know that the scaling are wrong and that one voxel in one image correspond to one voxel in one segmenttation.

So I tried to overwrite the spacing of my images:

```
my_img = tio.ScalarImage(img_path)
my_img.spacing = (1.0, 1.0, 1.0)
```

But then I get the bellow error:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_2544/1565851587.py in <module>
----> 1 my_img.spacing = (1.0, 1.0, 1.0)

AttributeError: can't set attribute
```

So here is my question: how can I overwrite the spacing on a `ScalarImage` and `LabelMap` ?

Thanks for the help!

**Comment by romainVala** [comment, marked as answer]
Hi
not sure  how to explicitely set the affine (that will define the spacing). may be @fepegar  has an idea how to achieve it ... ?

In the mean time you can try to  copy the affine from your image to you label with the transform `CopyAffine`
Thus you will have 2 coherent affine (and then spacing) but depending on the transform you used after this may not be enough (because having the correct affine is important, for instance  for Resampling )

**Reply by Optimox** [reply]
Thanks for your answer, I am a bit worried about unknown side effects...

**Comment by romainVala** [comment]
Actually you can trust dicom information, but I am not sure on the tiff one (this is not a volume format)
So copy the affine from the image (dicom) to replace the one from the label (in tiff, I guess)

**Reply by fepegar** [reply]
[TIFF images can contain a stack](http://soft-matter.github.io/pims/v0.5/tiff_stack.html#tiffstack). I agree that `CopyAffine` is probably the simplest solution here.
